File: 1726/1726.py
from itertools import accumulate

n = int(input())

pairs =  ( list( zip(*(list(map(int, input().split())) for i in range(n))))  )

pairs[0] = sorted(list(pairs[0]))
pairs[1] = sorted(list(pairs[1]))

cumsum_x = list(accumulate(pairs[0]))
cumsum_y = list(accumulate(pairs[1]))

sum = 0
for i in range(1, n):
    sum += -cumsum_x[i-1] + i*pairs[0][i]
    sum += -cumsum_y[i-1] + i*pairs[1][i]

print(sum//(n*(n-1)//2))

File: 1726/1726.rs
fn main () {
    let n: i64 = {
        let mut buf = String::new();
        std::io::stdin().read_line(&mut buf).expect("input");
        buf.trim().parse().unwrap()
    };
    
    let mut xs: Vec<i64> = Vec::new();
    xs.resize(10_000, 0);
    
    let mut ys: Vec<i64> = Vec::new();
    ys.resize(10_000, 0);

    for _ in 0..n {

        let pair: Vec<i64> = {
            let mut buf = String::new();
            std::io::stdin().read_line(&mut buf).expect("input");
            println!("`{buf}`");
            buf
                .trim()
                .split(" ")
                .flat_map(|s| {s.parse()})
                .collect()
        };

        xs.push(pair[0]);
        ys.push(pair[1]);
    }

    xs.sort();
    ys.sort();

    let mut cumsum_x: Vec<i64> = Vec::with_capacity(n as usize);
    let mut cumsum_y: Vec<i64> = Vec::with_capacity(n as usize);

    cumsum_x[0] = xs[0];
    cumsum_y[1] = ys[0];
    
    for i in 1..n as usize {
        cumsum_x[i] = xs[i] + cumsum_x[i-1];
        cumsum_y[i] = ys[i] + cumsum_y[i-1];
    }

    let sum = {
        let mut sum = 0;
        for i in 1..n as i64 {
            sum += i*xs[i as usize]-cumsum_x[(i-1) as usize];
            sum += i*ys[i as usize]-cumsum_y[(i-1) as usize];
        }
        sum
    };

    println!("{}", sum / (n*(n-1)/2) );

 
}